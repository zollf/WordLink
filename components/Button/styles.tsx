import { StyleSheet } from "react-native";

const styles = StyleSheet.create({
  button: {
    width: 175,
    height: 76,
    borderRadius: 5,
    display: "flex",
    justifyContent: "center",
    alignItems: "center",
    shadowColor: '#000',
    shadowRadius: 4,
    shadowOffset: { width: 2, height: 2},
    shadowOpacity: 0.1,
  },
  text: {
    fontSize: 18,
  }
});

export default styles;