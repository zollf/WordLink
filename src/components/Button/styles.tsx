import { StyleSheet } from 'react-native';
import { widthPercentageToDP, heightPercentageToDP } from 'react-native-responsive-screen';

import AppStyle from 'app/src/styles';

const styles = StyleSheet.create({
  button: {
    width: widthPercentageToDP('40%'),
    height: heightPercentageToDP('8%'),
    borderRadius: 5,
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    shadowColor: '#000',
    shadowRadius: 4,
    shadowOffset: { width: 2, height: 2 },
    shadowOpacity: 0.1,
  },
  text: {
    color: 'white',
    fontSize: heightPercentageToDP('2.5%'),
    fontFamily: AppStyle.fontBold,
  },
});

export default styles;
