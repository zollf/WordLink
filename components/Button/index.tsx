import React from 'react';
import { Text, TouchableOpacity, Animated } from 'react-native';
import { LinearGradient } from 'expo-linear-gradient';
import { useAnimate } from '../../hooks';

import AppStyle from '../../styles';
import styles from './styles';

interface Props {
  onPress: () => void;
  text: string;
  color?: 'green' | 'orange' | 'red' | 'blue' ;
}

const Button = ({ onPress, text, color }: Props) => {
  const getColor = () => {
    switch(color) {
      case 'green'  : return [AppStyle.lightGreen, AppStyle.green];
      case 'orange' : return [AppStyle.lightOrange, AppStyle.orange];
      case 'red'    : return [AppStyle.lightRed, AppStyle.red];
      default       : return [AppStyle.lightOrange, AppStyle.orange];
    }
  };

  const [spring] = useAnimate(1);
  const handlePress = () => {
    Animated.sequence([
      Animated.timing(spring, {
        toValue: 0.8,
        duration: 1,
        useNativeDriver: true,
      }),
      Animated.spring(spring, {
        toValue: 1,
        friction: 3,
        useNativeDriver: true,
      })
    ]).start();
    onPress();
  };

  return (
    <Animated.View style={{
      transform: [{scale: spring}]
    }}>
      <TouchableOpacity data-test-id="button" onPress={handlePress}>
        <LinearGradient colors={getColor()} style={styles.button} start={[-0.1, 0.1]} locations={[0.5, 0.5]}>
          <Text style={styles.text}>{text}</Text>
        </LinearGradient>
      </TouchableOpacity>
    </Animated.View>
  );
};

export default Button;