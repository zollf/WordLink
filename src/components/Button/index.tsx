import React from 'react';
import { Text, TouchableOpacity, Animated } from 'react-native';
import { LinearGradient } from 'expo-linear-gradient';
import { useAnimate } from 'app/src/hooks';

import { getGradient } from 'app/src/styles';
import styles from './styles';

interface Props {
  onPress: () => void;
  text: string;
  color?: string;
}

const Button = ({ onPress, text, color }: Props) => {
  const [spring] = useAnimate(1);
  const handlePress = () => {
    Animated.sequence([
      Animated.timing(spring, {
        toValue: 0.8,
        duration: 1,
        useNativeDriver: true,
      }),
      Animated.spring(spring, {
        toValue: 1,
        friction: 3,
        useNativeDriver: true,
      }),
    ]).start();
    onPress();
  };

  return (
    <Animated.View
      style={{
        transform: [{ scale: spring }],
      }}
    >
      <TouchableOpacity data-test-id="button" onPress={handlePress}>
        <LinearGradient colors={getGradient(color)} style={styles.button} start={[-0.1, 0.1]} locations={[0.5, 0.5]}>
          <Text style={styles.text}>{text}</Text>
        </LinearGradient>
      </TouchableOpacity>
    </Animated.View>
  );
};

export default Button;
