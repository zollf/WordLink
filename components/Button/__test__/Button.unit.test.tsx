import React from 'react';

import Button from '../';

import renderer from 'react-test-renderer';

const onPress = jest.fn();

const component = <Button text="button" color="green" onPress={onPress} />;

describe('<MenuBar/>', () => {
  const wrapper = renderer.create(component);
  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('click works correctly', () => {
    wrapper.root.findByProps({ 'data-test-id': 'button' }).props.onPress();
    expect(onPress).toBeCalled();
  });
});
