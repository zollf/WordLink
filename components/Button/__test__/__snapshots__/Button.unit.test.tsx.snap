// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuBar/> matches snapshot 1`] = `
<View
  style={
    Object {
      "transform": Array [
        Object {
          "scale": mockConstructor {},
        },
      ],
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <ViewManagerAdapter_ExpoLinearGradient
      proxiedProperties={
        Object {
          "colors": Array [
            4282711882,
            4280928549,
          ],
          "endPoint": undefined,
          "locations": Array [
            0.5,
            0.5,
          ],
          "startPoint": Array [
            -0.1,
            0.1,
          ],
        }
      }
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 5,
          "display": "flex",
          "height": 106.5,
          "justifyContent": "center",
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 2,
            "width": 2,
          },
          "shadowOpacity": 0.1,
          "shadowRadius": 4,
          "width": 300,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontFamily": "Raleway_700Bold",
            "fontSize": 33.5,
          }
        }
      >
        button
      </Text>
    </ViewManagerAdapter_ExpoLinearGradient>
  </View>
</View>
`;
